src/main/java/htsjdk/samtools/util/GZIIndex.java

```diff
@@ -221,31 +220,33 @@ public String toString() {
     /**
      * Writes this index into the requested path.
      *
+     * NOTE: This method will close out the provided output stream when it finishes writing the index
+     *
      * @param output the output file.
      *
      * @throws IOException if an I/O error occurs.
      */
-    public void writeIndex(final Path output) throws IOException {
+    public void writeIndex(final OutputStream output) throws IOException {
```

Comment:
My comment in the last round was about the fact that you changed the signature for this method, which was already public, which amounts to removing an existing method without first deprecating it. So although its probably correct for this new one be public, the old method that takes a Path needs to be restored, and changed to delegate to this one.